src/Symfony/Bundle/FrameworkBundle/HttpCache/HttpCache.php

```diff
@@ -83,6 +83,9 @@ protected function createSurrogate(): SurrogateInterface
 
     protected function createStore(): StoreInterface
     {
-        return $this->store ?? new Store($this->cacheDir ?: ($this->kernel->getShareDir() ?? $this->kernel->getCacheDir()).'/http_cache');
+        return $this->store ?? new Store($this->cacheDir ?: (match (true) {
+            null !== $this->kernel->getShareDir() => $this->kernel->getShareDir().'/cache/'.$this->kernel->getEnvironment().'/http_cache',
+            default => $this->kernel->getCacheDir().'/http_cache',
```

Comment:
let's also reduce nesting? not sure the `cache` level is worth it
what about moving the computation of cacheDir to the constructor also?

```php
$this->cacheDir = $cache ??= (null !== ($dir = $this->kernel->getShareDir()) ? $dir.'/'.$this->kernel->getEnvironment() : $this->kernel->getCacheDir()).'/http_cache';
```

```suggestion
        return $this->store ?? new Store($this->cacheDir);
```

Reply:
done